src/dmd/dmangle.d

```diff
@@ -1133,3 +1138,35 @@ extern (C++) void mangleToBuffer(TemplateInstance ti, OutBuffer* buf)
     v.mangleTemplateInstance(ti);
 }
 
+/******************************************************************************
+ * Mangle function signatures ('this' qualifier, and parameter types)
+ * to check conflicts in function overloads.
+ * It's different from fd.type.deco. For example, fd.type.deco would be null
+ * if fd is an auto function.
+ *
+ * Params:
+ *    buf = `OutBuffer` to write the mangled function signature to
+*     fd  = `FuncDeclaration` to mangle
+ */
+void mangleToFuncSignature(ref OutBuffer buf, FuncDeclaration fd)
+{
+    assert(fd.type.ty == Tfunction);
+    auto tf = cast(TypeFunction)fd.type;
+
+    scope Mangler v = new Mangler(&buf);
+
+    //if (fd.needThis() || fd.isNested())
+    //    buf.writeByte(Type.needThisPrefix());
+    MODtoDecoBuffer(&buf, tf.mod);
+    //switch (tf.linkage)
+    //{
+    //    case LINKd:         buf.writeByte('F');     break;
+    //    case LINKc:         buf.writeByte('U');     break;
+    //    case LINKwindows:   buf.writeByte('W');     break;
+    //    case LINKpascal:    buf.writeByte('V');     break;
+    //    case LINKcpp:       buf.writeByte('R');     break;
+    //    default:            assert(0);
```

Comment:
Remove these comments?

Reply:
Yeah they were part of the PR I revived and I wanted to port it as with minimal changes to have a good history, but now that I'm thinking about this: no one needs the history of a PR anyways, so I am learning to squash his work into an entire commit and maybe squash it entirely. Do we have any guidelines on this?

Reply:
It's possible to have separate authors and committers as well.

Reply:
Finally got around removing the comments. For simplicity I amended this to the Fixup commit (the one after Kenji's commits).